src/rt-app.c

```diff
@@ -759,6 +772,15 @@ void *thread_body(void *arg)
 				NULL);
 	}
 
+	/* TODO find a better way to handle that constraint:
+	 * Set the task to SCHED_DEADLINE as far as possible touching its
+	 * budget as little as possible for the first iteration.
+	 */
+
+	/* Set scheduling policy and print pretty info on stdout */
+	log_notice("[%d] Starting with %s policy with priority %d", data->ind, policy_to_string(data->sched_data->policy), data->sched_data->prio);
+	set_thread_priority(data, data->sched_data);
```

Comment:
right.. another try below :-)
where I'm try to exploit the fact that curr_sched_data is initialized to NULL and only set the first time we call set_thread_priority()
```
diff --git a/src/rt-app.c b/src/rt-app.c
index 856fe94..fe148eb 100644
--- a/src/rt-app.c
+++ b/src/rt-app.c
@@ -616,10 +616,11 @@ static void set_thread_affinity(thread_data_t *data, cpuset_data_t *cpu_data)
 	}
 }
 
-static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
+static void set_thread_priority(thread_data_t *data, sched_data_t *psched_data)
 {
 	struct sched_param param;
 	policy_t policy;
+	sched_data_t *sched_data = psched_data;
 #ifdef DLSCHED
 	struct sched_attr dl_params;
 	pid_t tid;
@@ -627,8 +628,12 @@ static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
 #endif
 	int ret;
 
-	if (sched_data == NULL)
-		return;
+	if (sched_data == NULL) {
+		if (data->curr_sched_data == NULL)
+			sched_data = data->sched_data;
+		else
+			return;
+	}
 
 	if (data->curr_sched_data == sched_data)
 		return;
@@ -808,8 +813,10 @@ void *thread_body(void *arg)
 	 */
 
 	/* Set scheduling policy and print pretty info on stdout */
-	log_notice("[%d] Starting with %s policy with priority %d", data->ind, policy_to_string(data->sched_data->policy), data->sched_data->prio);
-	set_thread_priority(data, data->sched_data);
+	//log_notice("[%d] Starting with %s policy with priority %d",
+	//	   data->ind, policy_to_string(data->sched_data->policy),
+	//	   data->sched_data->prio);
+	//set_thread_priority(data, data->sched_data);
 
 	/*
 	 * phase        - index of current phase in data->phases array
```